### PR TITLE
feat: added reusable workflows for terraform

### PR DESCRIPTION
**Pull request by amartin-isp**
This PR adds 3 reusable workflows:
- `terraform`: this is the classic `terraform` workflow that is used by `kubernetes_management` and other repos for terraform related pr maintenance and actions.
- `terraform_docs`: this workflow automatically generates module docs in the specified output file, and optionally commit the updates to the PR.
- `terraform_fmt`: this workflow automatically runs `terraform fmt`, and optionally commit the updates to the PR.

### CODE REVIEW

**Comment by amartin-isp** [comment]
I can split this PR into the separate workflows, but since I decided against it since its a common theme (new reusable terraform workflows).

File: .github/workflows/terraform.yml [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+          cp ./.tool-versions $HOME/
+          asdf plugin-add terraform
+          asdf plugin-add sops
+          asdf install
+      - name: "terraform fmt"
+        uses: iStreamPlanet/github-actions/terraform-actions@v1.13.0
+        if: github.event_name == 'pull_request'
+        with:
+          command: fmt
+          working_directory: ${{ env.WORKSPACE }}
```
[the remark this answers is not in the document]
Comment (amartin-isp) [reply .github/workflows/terraform.yml:94]: Yea, thats why I made the comment above. The difference here is that this fmt just checks if the module is formated correctly and errors if its not, whereas the workflow will perform the action and push the updates.

File: .github/workflows/terraform_docs.yml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    steps:
+      - name: checkout
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+
+      - name: module terraform-docs
+        uses: terraform-docs/gh-actions@v1.0.0
+        with:
+          working-dir: "${{ matrix.workspace }}"
```
[the remark this answers is not in the document]
Comment (amartin-isp) [reply .github/workflows/terraform_docs.yml:75]: done

File: .github/workflows/terraform.yml [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.aws_secret_access_key }}
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      WORKSPACE: clusters/aws/${{ matrix.workspace }}
+      WORKFLOW_RUN_URL: https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install asdf on the Action Runner
+        uses: asdf-vm/actions/setup@v1.1.0
+        with:
+          asdf_branch: "v0.8.0"
```
Comment (marcind) [inline .github/workflows/terraform.yml:81]:
```suggestion
          asdf_branch: "v0.10.2"
```
Probably can move to the latest: https://github.com/asdf-vm/asdf/releases

**Review by isnellfeikema-isp — APPROVED** [review]
LGTM.

**Pull request closed without merging**